src/librustc/ty/error.rs

```diff
@@ -144,8 +144,7 @@ impl<'tcx> fmt::Display for TypeError<'tcx> {
                            found bound lifetime parameter {}", br)
             }
             Sorts(values) => ty::tls::with(|tcx| {
-                report_maybe_different(f, values.expected.sort_string(tcx),
-                                       values.found.sort_string(tcx))
+                write!(f, "expected {}", values.expected.sort_string(tcx))
```

Comment:
That's totally fine (in fact, that's the whole reason why I have a separate commit removing all the "found ___" part). It does make me feel like it makes more sense to change all of them at once. Not sure if this interim state is desirable though, i.e. having the span label state "expected ___, found ____" instead of simply "expected _____".